c/datasetjson.c

```diff
@@ -2501,9 +2643,120 @@ bool checkIfDatasetExists(char* dataset) {
   SLHFree(slh);
 
   if(datasetCount > 0) {
-    return true;
+    // respondWithJsonError(response, "Target dataset already exists", 400, "Bad Request");
+    // return ERROR_DATASET_ALREADY_EXIST;
+    return 1;
+  }
+  return 0;
+}
+
+void pasteAsDatasetMember(HttpResponse *response, char* sourceDataset, char* targetDataset) {
+  bool isTargetMember = true;
+
+  int reasonCode = 0;
+  int rc = createDataset(response, targetDataset, NULL, 0, &reasonCode);
+  if(rc != 0) {
+    return;
+  }
+
+  char msgBuffer[128];
+  char etag[128];
+
+  rc = readWriteToDatasetAndRespond(response, sourceDataset, targetDataset, isTargetMember, msgBuffer, etag);
+  if(rc >= 0) {
+    jsonPrinter *p = respondWithJsonPrinter(response);
+    setResponseStatus(response, 201, "Successfully Copied Dataset");
+    setDefaultJSONRESTHeaders(response);
+    writeHeader(response);
+    jsonStart(p);
+
+    jsonAddString(p, "msg", msgBuffer);
+    jsonAddString(p, "etag", etag);
+
+    jsonEnd(p);
+    finishResponse(response);
+  }
```

Comment:
if rc < 0, you still need to respond right? be sure that either it already did respond, or that you do it here.

Reply:
If rc<0, we respond in the function` readWriteToDatasetAndRespond`